src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -80,6 +80,21 @@ SolverInterfaceImpl::SolverInterfaceImpl(
   logging::setParticipant(_accessorName);
 
   configure(configurationFileName);
+
+// This block cannot be merge with the one above as only configure calls
+// utils::Parallel::initializeMPI, which is needed for getProcessRank.
+#ifndef PRECICE_NO_MPI
+  if (communicator != nullptr) {
```

Comment:
Why not check it straight after setting the communicator here?
https://github.com/precice/precice/blob/3699cc0b3ba1ed6a907a86abc97367df3389584f/src/precice/impl/SolverInterfaceImpl.cpp#L70-L78